### PR TITLE
Updates for contributor plus (& more)

### PR DESCRIPTION
**Pull request by Beamanator**
@mallenexpensify requesting you review for C+ reasons :D

### Details
Updating documentation a lil bit - fixing some grammar & adding references to C+ where needed.

### Fixed Issues
N/A

### CODE REVIEW

File: CONTRIBUTING.md
Comment (jasperhuangg) [inline CONTRIBUTING.md:5]: `app philosophy` is kinda ambiguous here. I'd just say `best practices` maybe (unless you wanted something more general).

File: CONTRIBUTING.md
Comment (jasperhuangg) [inline CONTRIBUTING.md:8]:
```suggestion
You can create as many accounts as needed in order to test your changes directly from [the app](https://new.expensify.com/). An initial account can be created when logging in for the first time, and additional accounts can be created by using the "New Chat" or "Group Chat" option in the Global Create Menu, inputting a valid email or phone number , and tapping the user's avatar.
```

Comment (jasperhuangg) [reply CONTRIBUTING.md:8]: idk if `Global Create Menu` is a widely accepted term (referring to the floating green `+` button), but I think this explanation is a bit more succinct imo.

File: CONTRIBUTING.md
Comment (jasperhuangg) [inline CONTRIBUTING.md:49]:
```suggestion
4. After review in #expensify-open-source, if you've provided a quality proposal that we choose to implement, a GitHub issue will be created and your Slack handle will be included in the original post after `Issue reported by:`
```

File: README.md
Comment (jasperhuangg) [inline README.md:228]:
```suggestion
- If you export a constant named `CONST`, its file/directory should be named the `CONST`.
- If you export a component named `Text`, the file/directory should be named `Text`.
- If you export a function named `guid`, the file/directory should be named `guid`.
```

File: README.md
Comment (jasperhuangg) [inline README.md:230]: Not gonna lie, I had no idea what an HOC was when I first got to Expensify, especially since I was using functional components and hooks (which I assume to be the norm with React developers these days). 

Any chance we can link to some docs + an example in our code here? Also, perhaps spell out `Higher-Order Components` the first time it's referenced.

**Review by jasperhuangg — CHANGES_REQUESTED** [review]
Few small changes, noiceee

File: CONTRIBUTING.md
Comment (jasperhuangg) [reply CONTRIBUTING.md:5]: ahhh I like this more! App philosophy seems like it encompasses those other facets.

File: CONTRIBUTING.md
Comment (Beamanator) [reply CONTRIBUTING.md:8]: +1 -> I kinda like this phrasing even better:

> ... and additional accounts can be created by opening the "New Chat" or "Group Chat" pages via the Global Create menu, inputting a valid email or phone number, and tapping the user's avatar.

File: CONTRIBUTING.md
Comment (Beamanator) [reply CONTRIBUTING.md:49]: Nice 👍

File: README.md
Comment (Beamanator) [reply README.md:228]: Agreed

File: README.md
Comment (Beamanator) [reply README.md:230]: Good idea - I think we could just link to [this page](https://reactjs.org/docs/higher-order-components.html) for "what are HOCs", and maybe this whole point could look like:

> - [Higher-Order Components](https://reactjs.org/docs/higher-order-components.html) (HOCs) should be named in camelCase, like `withOnyx` (example [here](https://github.com/Expensify/App/blob/08a9b226a27a6f24bb11c3006cf433739c834860/src/Expensify.js#L214-L231)).

Comment (Beamanator) [reply README.md:230]: Hmm on second thought I don't think we need the `(example here)` part, if we call out `withOnyx` it should be pretty easy to search for examples right?

File: CONTRIBUTING.md
Comment (jasperhuangg) [reply CONTRIBUTING.md:8]: Awesome, I think my explanation was a bit unclear there, but this one makes a ton of sense!

File: README.md
Comment (jasperhuangg) [reply README.md:230]: I dig!!

File: CONTRIBUTING.md
Comment (Beamanator) [reply CONTRIBUTING.md]: Nice, agreed!

File: CONTRIBUTING.md
Comment (Beamanator) [reply CONTRIBUTING.md]: Nice! Team work 👍

**Comment by Beamanator** [comment]
Updated! Thanks for the review @jasperhuangg ! 💯

**Review by jasperhuangg — DISMISSED** [review]
Looks good! Let's wait on the other review before merging.

**Comment by mallenexpensify** [comment]
LGTM but unsure about this
image

The contributor will technically need to apply to the job in Upwork in order to get paid.  But... I think I see what you're getting at
- Make sure you have Upwork account
- Apply in GH
- If hired/assigned in GH _then_ the contributor will be directed to the specific job and they can apply there then (cuz applying there doesn't matter unless they've been hired in GH)

Is that correct?

**Comment by Beamanator** [comment]
Yeah @mallenexpensify that's exactly right, I was hoping we could make it more clear that they technically apply in GH first, _then_ Upwork (once they're hired via GH) - so we can probably improve that change right?

**Review by mallenexpensify — DISMISSED** [review]
GTG!

File: CONTRIBUTING.md
Comment (Beamanator) [inline CONTRIBUTING.md:30]: @mallenexpensify how does this phrasing sound for this section?

Comment (mallenexpensify) [reply CONTRIBUTING.md:30]: I like it!

**Review by mallenexpensify — APPROVED** [review]

**Pull request merged**
